### ISSUE TITLE
image_default_link_type is ignored by Gutenberg

### ISSUE THREAD

**Issue by Xoliul** [WordPress/gutenberg#10173]
The setting image_default_link_type  from options.php is not used by Gutenberg: mine is set to "file", yet the editor defaults every image or gallery to the "none" link type. I have to set them all manually instead.

**Comment by TomasValenta** [comment]
Please fix this annoying bug :(

**Comment by TomasValenta** [comment]
Finally, I found this working replacement!

```php
function gallery_template_to_posts() {
    $post_type_object = get_post_type_object( 'post' );
    $post_type_object->template = array(
        array( 'core/gallery', array(
            'linkTo' => 'media',
        ) ),
    );
}
add_action( 'init', 'gallery_template_to_posts' );
```

**Comment by Xoliul** [comment]
> Finally, I found this working replacement!
> 
> ```
> function gallery_template_to_posts() {
>     $post_type_object = get_post_type_object( 'post' );
>     $post_type_object->template = array(
>         array( 'core/gallery', array(
>             'linkTo' => 'media',
>         ) ),
>     );
> }
> add_action( 'init', 'gallery_template_to_posts' );
> ```

Any reference what file this is changed in?

**Comment by TomasValenta** [comment]
For example `functions.php` in the child theme folder ;)
https://codex.wordpress.org/Functions_File_Explained

**Comment by trlanfeng** [comment]
I found that the 'Link To' default setting is write in the `wp-includes/js/dist/block-library(min).js`  file at Line 3584.
```
linkDestination: {
    type: 'string',
    default: 'none'
},
```
After I change the default to 'media' , every image that I insert into the post , is 'Link To Media File' as default now.
But I don't know how to change the default value in the 'function.php' file

**Comment by TomasValenta** [comment]
You have a working example three post above ;)

**Comment by PTArtist** [comment]
> 
> 
> Apparently, this is a case of "your mileage may vary."
> 
> When I added this code to functions.php, a gallery block was automatically added to new posts and there was no change to link to.
NOTE there are two different suggestions for adding to functions.php via child theme... Did you use the one that adds the set link function:     "function set_link_to_media_by_default"   or the other one?

**Comment by PTArtist** [comment]
I tried creating a new tiny child theme, to define the above "function set_link_to_media_by_default()", and after some confusion, was able to successfully activate my child theme, BUT new galleries still default to link "none". And I find that by using a child theme I lose all my many randomized header files, and have re-establish them as header files (from medial library) ONE at a TIME rather than by selecting multiple filles... lame!

**Comment by HmCody** [comment]
> > NOTE there are two different suggestions for adding to functions.php via child theme... Did you use the one that adds the set link function:     "function set_link_to_media_by_default"   or the other one?

Sorry for the confusion. The set link function was the one I tried.

**Comment by SuzuKube** [comment]
I think we should have an option for default Media Link on Gallery. The pictures are ridiculously little O_o !

**Comment by iandunn** [comment]
Also [reported on the forums](https://wordpress.org/support/topic/a-chier/). I was able to reproduce it.

**Comment by erfanio** [comment]
It's been a while since the last update. Is there a timeline when this going to be fixed?

**Comment by Furiouss38** [comment]
> 
> 
> Finally, I found this working replacement! Put this code to the `functions.php` file:
> 
> ```
> <?php
> 
> add_action( 'init', 'set_link_to_media_by_default' );
> function set_link_to_media_by_default() {
>     $post_type_object = get_post_type_object( 'post' );
>     $post_type_object->template = array(
>         array( 'core/gallery', array(
>             'linkTo' => 'media',
>         ) ),
>     );
> }
> ```
> 
> You can also specify another attributes too, for example columns:
> 
> ```
> <?php
> 
> add_action( 'init', 'set_link_to_media_by_default' );
> function set_link_to_media_by_default() {
>     $post_type_object = get_post_type_object( 'post' );
>     $post_type_object->template = array(
>         array( 'core/gallery', array(
>             'linkTo' => 'media',
>             'columns' => 4,
>         ) ),
>     );
> }
> ```

When I add this code to functions.php, it's like HmCody said. 
Links still defined to none and a gallery object is added in each new post.

It's annoying to have to set the link for each image and gutenberg not using image_default_link_type

**Comment by noisysocks** [comment]
This is still awaiting development. The block editor needs to respect the existing `image_default_link_type` filter that WordPress has.

**Comment by helloiamgergo** [comment]
Any updates? This feature (or we shall call it basic functionality...) would be crucial in my project.
I need this to work with Media & Text block. I tried @TomasValenta 's solution. I modified "core/gallery" to "core/media-text" but it's no use. I also tried the "core/image" as Media & Text is based on it. Do you have any ideas?

@WordPress: This issue exists for a long time. Is that so difficult to fix such a small bug?

**Comment by idarek** [comment]
The above solution by adding code to functions.php doesn't work for.
Firstly, it adding gallery block to every new post, which is unneeded.

Every other gallery block is still with none by default. 
Also, how about past-posts? changing them from none to media?

**Comment by vigilantspb** [comment]
any news? it's so annoying :(

**Comment by web-jst** [comment]
I tried the following, which seems to work.

Like in [this example](https://developer.wordpress.org/block-editor/developers/filters/block-filters/#block-style-variations), you can register a Gutenberg JavaScript file `customize_block.js` in `functions.php`:
```
function myguten_enqueue() {
    wp_enqueue_script(
        'myguten-script',
        get_template_directory_uri() .  '/customize_block.js',
        array( 'wp-blocks', 'wp-dom-ready', 'wp-edit-post' ),
        filemtime(get_template_directory_uri() . '/customize_block.js')
    );
}
add_action('enqueue_block_editor_assets', 'myguten_enqueue');
```

Similar to [this example](https://developer.wordpress.org/block-editor/developers/filters/block-filters/#blocks-registerblocktype), you can now change the default settings of blocks in `customize_block.js`:
```
function changeBlockSettings(settings, name) {
    if (name == 'core/gallery') {
        settings.attributes.linkTo.default = 'media';
    }
    return settings;
}

wp.hooks.addFilter(
    'blocks.registerBlockType',
    'yourNamespace',
    changeBlockSettings
);
```
The possible settings of gallery blocks can be found [here](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/gallery/block.json).
In the same way, I was also able to change the settings of [other block types](https://github.com/WordPress/gutenberg/tree/master/packages/block-library/src).

**Comment by goaround** [comment]
The problem with this approche is that the settings.attributes.linkTo.default value will not store inside the block markup because it's the default. If you change the default value later or remove the filter, the block will break.

**Comment by web-jst** [comment]
Thanks for pointing that out.
I have now experimented with removing the filter. After removing it, the gallery block was still correctly displayed on the website but in the editor I got an "Unexpected or Invalid Content" error. So you can't modify the gallery anymore with the filter deactivated.

**Comment by mkaz** [comment]
We just fixed this in #25578 and #25582, it will go out in Gutenberg 9.1 and you can test then, or if you are setup to test from the repository both are merged in.
